RELEASE_CHECKLIST.md

```diff
@@ -1,97 +1,168 @@
 # Release Checklist
 
-This is a list of the things that need to happen during a release.
+This is a list of the things that need to happen during a release. We support three kinds of releases:
+1. A standard release, from `main` (i.e. v0.27.0, v0.28.3, v1.5.2, etc.)
+2. A release candidate release, from `main` (i.e. v0.27.0-rc.0, v1.5.3-rc.3, etc.)
+3. A pre-release release, from any branch (i.e. v0.26.0-PR1234, v0.27.4-my.build.0)
 
-## Build a Release
+_N.B All the version numbers above follow Semantic Versioning, and specifically its notion of encoding pre-release information after the `-` for more information see the [Semantic Versioning specification](https://semver.org/)_
 
-### Prepare the Changelog (Full release only)
+## Standard Release
 
-If you are releasing a beta or a release candidate, no official changelog is needed, but you're not off the hook! You'll need to write testing instructions in lieu of an official changelog.
+As noted above, a standard release takes the form of a tagged commit on the `main` branch, and includes all
+commits previous to it. Preparing it involves several phases:
+
+### Prepare The Changelog
```

Comment:
is this just for the standard release or for bespoke ones too? if so, wondering about the vNext milestone handling

Reply:
oh, it totally is, I just wasn't looking at the headers